### ISSUE TITLE
Suggestion: don't check that a file exists before operating on it

### ISSUE THREAD

**Issue by hongooi73** [rstudio/pins#512]
I've noticed that in several places, pins will try to verify that a file/directory exists before doing stuff with it:

https://github.com/rstudio/pins/blob/b5696d1909c9de323a1532bede2fa0a7e84a3e80/R/board_s3.R#L150-L153
https://github.com/rstudio/pins/blob/b5696d1909c9de323a1532bede2fa0a7e84a3e80/R/board_azure.R#L71-L72
https://github.com/rstudio/pins/blob/b5696d1909c9de323a1532bede2fa0a7e84a3e80/R/board_folder.R#L98-L100

Instead of this, consider simply going ahead with the operation, and throwing an error if it fails. It sounds counterintuitive, but checking for existence is actually an anti-pattern: https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use

Basically, there is a race condition: in between the time that you do the check, and the time that you do the operation, someone else might have deleted/modified the file. This is probably less important for pins than for big databases, but maybe still something to consider.

**Comment by hadley** [comment]
The problem is that the errors tend not to be very user unfriendly, and don't immediately reveal what the problem is.
